modules/launcher/src/main/java/org/mule/module/launcher/artifact/DefaultResourceReleaser.java

```diff
@@ -32,6 +34,37 @@
     public void release()
     {
         deregisterJdbcDrivers();
+
+        cleanUpResourceBundle();
+    }
+
+    private void cleanUpResourceBundle()
+    {
+        try
+        {
+            ResourceBundle.clearCache(this.getClass().getClassLoader());
+        }
+        catch(Exception e)
+        {
+            logger.warn("Couldn't clean up ResourceBundle", e);
+        }
+
+        try
+        {
+            // NONEXISTENT_BUNDLE leaves a cacheKey referencing to this classloader
+            Field nonExistentBundleField = ResourceBundle.class.getDeclaredField("NONEXISTENT_BUNDLE");
+            nonExistentBundleField.setAccessible(true);
+
+            ResourceBundle resourceBundle = (ResourceBundle) nonExistentBundleField.get(null);
+
+            Field cacheKeyField = ResourceBundle.class.getDeclaredField("cacheKey");
+            cacheKeyField.setAccessible(true);
+            cacheKeyField.set(resourceBundle, null);
+        }
+        catch (Exception e)
+        {
+            logger.warn("Couldn't clean up reference from NONEXISTENT_BUNDLE on ResourceBundle", e);
```

Comment:
NONEXISTENT_BUNDLE is not something the user must know about. Message should say something like "Unable to clean up ResourceBundle references, this can cause a memory leak" or something like that